sled-agent/src/services.rs

```diff
@@ -927,11 +927,34 @@ impl ServiceManager {
         needed
     }
 
+    /// Enable DNS resolution using the internal DNS servers for the given zone.
+    ///
+    /// The Internal DNS servers are allocated on a reserved AZ-wide subnet which we can
+    /// determine from the given Sled underlay address.
+    ///
+    /// For more details on this "reserved" rack subnet, refer to
+    /// [omicron_common::address::ReservedRackSubnet].
+    async fn configure_internal_dns_in_zone(
```

Comment:
This could be a boon for debugging internal services, even if we choose to rely on a non-host resolver at times.

Reply:
Yea, but perhaps this specific method wouldn't be as useful when everything has moved to be self-assembling.